src/gt4py/next/common.py

```diff
@@ -52,10 +52,14 @@ def __str__(self):
         return f"{type(self).__name__}.{self.name}"
 
 
-@dataclass(frozen=True)
+@dataclasses.dataclass(frozen=True)
 class Dimension:
     value: str
-    kind: DimensionKind = DimensionKind.HORIZONTAL
+    kind: DimensionKind = dataclasses.field(
+        default=DimensionKind.HORIZONTAL,
+        hash=False,
+        compare=False,  # TODO(havogt): include kind in hash and comparison, disabled because for itir dimension is just the tag
+    )
```

Comment:
If you think this should be immediately removed, I'd propose to discuss if we should extend Iterator IR with this information...